spring-cloud-sleuth-brave/src/main/java/org/springframework/cloud/sleuth/brave/autoconfig/TraceBraveAutoConfiguration.java

```diff
@@ -83,84 +82,78 @@
 	 */
 	public static final String DEFAULT_SERVICE_NAME = "default";
 
-	@Bean
-	@ConditionalOnMissingBean
-	BraveBaggageManager braveBaggageManager() {
-		return new BraveBaggageManager();
-	}
-
-	@Bean(name = TRACING_BEAN_NAME)
-	@ConditionalOnMissingBean
-	// NOTE: stable bean name as might be used outside sleuth
-	Tracing tracing(@LocalServiceName String serviceName, Propagation.Factory factory,
-			CurrentTraceContext currentTraceContext, Sampler sampler, SleuthProperties sleuthProperties,
-			@Nullable List<SpanHandler> spanHandlers, @Nullable List<TracingCustomizer> tracingCustomizers) {
-		Tracing.Builder builder = Tracing.newBuilder().sampler(sampler)
-				.localServiceName(StringUtils.isEmpty(serviceName) ? DEFAULT_SERVICE_NAME : serviceName)
-				.propagationFactory(factory).currentTraceContext(currentTraceContext)
-				.traceId128Bit(sleuthProperties.isTraceId128()).supportsJoin(sleuthProperties.isSupportsJoin());
-		if (spanHandlers != null) {
-			for (SpanHandler spanHandlerFactory : spanHandlers) {
-				builder.addSpanHandler(spanHandlerFactory);
+	@Configuration(proxyBeanMethods = false)
+	static class BraveConfiguration {
```

Comment:
Why do you need a nested class here? It doesn't look necessary considering there aren't any condition on it.